src/main/java/com/xeeva/catalog/CostCenterPage.java

```diff
@@ -112,30 +135,122 @@ public void AddInternalComment(String InternalComment){
+	 * @date    20/09/16
+	 */
+	/**
+	 * @param verifyType
+	 * @param itemNumber
+	 * @param CCValue
+	 * @param QuantityValue
+	 * @return
 	 */
-	public boolean verifyCC_LineLevel(String itemNumber,String CCValue) {
-		boolean statusflag = false;
+	public boolean verifyCostCenter(String verifyType,String itemNumber,String CCValue,String QuantityValue){
+		boolean statusFlag = false;
 		List<WebElement> readLinks = driver.findElements(By.xpath(".//*[@id='customfa']/tbody/tr/td[1]/a"));
-		for(WebElement inputLink : readLinks){
-			List<WebElement> readSelects = driver.findElements(By.xpath(".//*[@id='customfa']/tbody/tr/td/select"));
-			for(WebElement inputSelects: readSelects){
-				String updatedCC = new Select(inputSelects).getFirstSelectedOption().getText();
-				if(inputLink.getText().equalsIgnoreCase(itemNumber) &&updatedCC.equalsIgnoreCase(CCValue)){
-					statusflag = true;			
-					break;
+		List<WebElement> readSelects = driver.findElements(By.xpath(".//*[@id='customfa']/tbody/tr/td/select"));
+		List<WebElement> readDeleteIcons = driver.findElements(By.xpath(".//*[@title='Delete']"));
+
+		switch(verifyType.toLowerCase()){
+		case "linelevel":
+			for(WebElement inputLink : readLinks){
+				for(WebElement inputSelects: readSelects){
+					String updatedCC = new Select(inputSelects).getFirstSelectedOption().getText();
+					if(inputLink.getText().equalsIgnoreCase(itemNumber) &&updatedCC.equalsIgnoreCase(CCValue)){
+						statusFlag = true;			
+						break;
+					}
+				}
+			}
+
+			break;
+
+		case "headerlevel": 
+			for(int i=0;i<readSelects.size();i++){
+				String ExpectedCC = new Select(readSelects.get(i)).getFirstSelectedOption().getText();
+				if(ExpectedCC.equalsIgnoreCase(CCValue)){
+					statusFlag =true;
+				}
+			}
+			break;
+
+		case "copyitem":
+			for(WebElement input :readSelects){
+				if(new Select(input).getFirstSelectedOption().getText().contains(CCValue)){
+					String randomID =input.getAttribute("id").replaceAll("\\D+", "");
+					WebElement element = driver.findElement(By.xpath(".//*[@id='trCCRow_"+randomID+"']/td[9]"));
+					driver.executeJavaScript("arguments[0].click();", element);
+					if(txtShowFullComment.isDisplayed()){
+						TestReporter.assertTrue(false, "Copied Item having Internal Comments");
+					}else{
+						TestReporter.assertTrue(true, "Copied Item is not having Internal Comments");
+					}
 				}
 			}
+			break;
+
+		case "quantity": 
+			// Change Quantity - Line Level record
+			changeQuantity(QuantityValue,lstSave.get(0));
+			break;
+		case "savecart": 
+			// Verify Save Cart 
+			changeQuantity(QuantityValue,btnSaveCart);
+			break;
+
+		case "delete":
+			// Perform Delete 
+			int sizeBeofreDelete =lstQty.size();
+			driver.executeJavaScript("arguments[0].click();", readDeleteIcons.get(0));
+			//Handle Alert if present
+			if(AlertHandler.isAlertPresent(driver, 6)){
+				AlertHandler.handleAlert(driver, 6);
+			}
+			// Verify Delete 
+			Sleeper.sleep(9000);

```

Comment:
This is a long hard wait, is there another way we can wait for the delete to finish?